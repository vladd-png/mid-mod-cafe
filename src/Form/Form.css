.user-form {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin-bottom: 50px;
}

.form-input {
  width: 200px;
  font-size: 16px;
  margin: 10px;
}

.submit-btn {
  background-color: #028080;
  font-size: 16px;
  color: #FFF;
  height: 26px;
  margin-top: 8px;
  width: 180px;
}
