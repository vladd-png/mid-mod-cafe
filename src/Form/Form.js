import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      date: '',
      time: '',
      number: null
    }
  }
  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value })
  }
  render() {
    return(
      <form className='user-form'>
        <input className='form-input' type='text' placeholder='Name' value={this.state.name} onChange={this.handleChange} name='name' />
        <input className='form-input' type='text' placeholder='Date (mm/dd)' value={this.state.date} onChange={this.handleChange} name='date' />
        <input className='form-input' type='text' placeholder='Time' value={this.state.time} onChange={this.handleChange} name='time' />
        <input className='form-input' type='text' placeholder='Number of Guests' value={this.state.number} onChange={this.handleChange} name='number' />
        <button className='submit-btn' type='button'>Make Resservation</button>
      </form>
    )
  }
}

export default Form
