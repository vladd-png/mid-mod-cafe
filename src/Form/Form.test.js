import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Form />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Form', () => {
  let wrapper;
  let mockAddReservation = jest.fn()

  beforeEach(() => {
    wrapper=shallow(<Form
        addReservation={mockAddReservation}
      />);
  })

  it('should be an instance of Form', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when addReservation is invoked', () => {
    wrapper.find('button').simulate('click');
    expect(mockAddReservation).toHaveBeenCalled();
  })

})
