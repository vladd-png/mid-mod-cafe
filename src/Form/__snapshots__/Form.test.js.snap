// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should be an instance of Form 1`] = `
<form
  className="user-form"
>
  <input
    className="form-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="form-input"
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    className="form-input"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    className="form-input"
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="text"
    value=""
  />
  <button
    className="submit-btn"
    onClick={[Function]}
    type="button"
  >
    Make Resservation
  </button>
</form>
`;
